### DISCUSSION TITLE
Commvault Backup doesn't contain data, can't scan directories below first level

### DISCUSSION THREAD

**Discussion by kirihst** [minio/minio#20093] [Q&A] [answered]
Hello team,

In commvault we configured a cloud apps backupclient according to the documentation at https://blog.min.io/commvault-object-storage/ to backup s3 buckets. Fairly recently we discovered, that the backupjobs of those clients are not writing any data to disk anymore. All backups are 0 bytes to 2kb in size while the buckets with the source data is at least a few gigabytes. there are mostly no errors during backup. I suspected there being an issue with user permissions, as I was giving commvault the minio root user credentials of an instance, but changing the user to another capable user did not change the result. 

Viewing the subclient properties from the commvault console, I can even browse the buckets, but no data from inside those buckets is available.

I collected a few logs from the commvault jobs. please see the following errors:

>> Selected Contents are

8988  1228 07/08 10:51:15 7130695 CSubclientContentCloud::LoadScanContent(344) - content[/]
8988  1228 07/08 10:51:15 7130695 CSubclientContentCloud::LoadScanContent(344) - content[/backup]
8988  1228 07/08 10:51:15 7130695 CSubclientContentCloud::LoadScanContent(344) - content[/bor-tsdb]
8988  1228 07/08 10:51:15 7130695 CSubclientContentCloud::LoadScanContent(344) - content[/is-backup]
8988  1228 07/08 10:51:15 7130695 CSubclientContentCloud::LoadScanContent(344) - content[/packer]
8988  1228 07/08 10:51:15 7130695 CCloudWorkUnitManager::AddScanPathToWorkUnit(244) - WorkUnit=0x00000294CDE3F940, PhysicalMediaId=cloud-physical-media-id, LocalVolumeUniqueId=cloud-vol-uniqueid, LocalVolumeName=\\CloudDummyPrefix, LocalScanPath=\\CloudDummyPrefix\is-backup
8988  1228 07/08 10:51:15 7130695 CScanJobCloud::Run(883) - ScanContent - '[\\CloudDummyPrefix\packer](file:///packer)'
8988  1228 07/08 10:51:15 7130695 GetRootFolderStatus() - Error: 400 Bad Request.
8988  1228 07/08 10:51:15 7130695 CVBaseRemoteFileWrapper::GetCloudFolderStatus(476) - Failed to get Cloud folder status. Error[44105][400 Bad Request.]
8988  1228 07/08 10:51:15 7130695 CCloudStorageFS::GetFileInfo(179) - GetCloudFolderStatus failed for[packer]. Error[0xEC02AC49:{CVBaseRemoteFileWrapper::GetCloudFolderStatus(478)/MM.44105-400 Bad Request.}]

8988  4570 07/08 10:51:15 7130695 CScanTraverseFolderPrivate::TraverseFolder(653) - CVFindFirstFile failed, ignoring, SearchPath=\\CloudDummyPrefix\backup\*, error=0xECCC0000:{CCloudPathEnumerator::FindFirst(165)/ErrNo.0.(No error)-ListDirectory(\\CloudDummyPrefix\backup\*) failed}
8988  4570 07/08 10:51:15 7130695 The authorization header is malformed; the region is wrong; expecting 'BOREUS'. Resource: /backup. Details: BucketName = "backup" Region = "BOREUS".
8988  4570 07/08 10:51:15 7130695 CVBaseRemoteFileWrapper::GetCloudFileList(455) - Failed to get Cloud file list. Error[44058][The authorization header is malformed; the region is wrong; expecting 'BOREUS'. Resource: /backup. Details: BucketName = "backup" Region = "BOREUS".]
8988  4570 07/08 10:51:15 7130695 CCloudFSwithPaging::ListDirectoryInternalPageWise(778) - GetFileList failed for[backup].Error[-335369190][0xEC02AC1A:{CVBaseRemoteFileWrapper::GetCloudFileList(457)/MM.44058-The authorization header is malformed; the region is wrong; expecting 'BOREUS'. Resource: /backup. Details: BucketName = "backup" Region = "BOREUS".}]
8988  4570 07/08 10:51:15 7130695 CCloudStorageFS::ListDirectory(284) - GetFileList failed for[backup].Error[0][Success]

8988  1228 07/08 10:51:13 7130695 [CVCloudBkpNew] <= x-content-type-options: nosniff
8988  1228 07/08 10:51:13 7130695 [CVCloudBkpNew] <= x-minio-error-code: AuthorizationHeaderMalformed

Our Minio version is 2024-03-26T22:10:45Z, Commvault Version is 11.32.49

I tried creating a new minio instance to rule out previous configuration changes as the cause. Backing up the new instance resulted in the following events during backup

![grafik](https://github.com/user-attachments/assets/007902e4-08fc-47a2-bc22-f0b8455f0c0d)

Commvault support couldn't find the issue but gave me the following information:
--
>> With the shared log, we found that the following API request were failing

GetRootFolderStatus
GetFileInfo
GetCloudFolderStatus

And only the metadata of the folders were backed up.
https://min.io/docs/minio/linux/developers/java/API.html
Solution/Fix
>> The root cause is on the S3 end and needs to be validated further from AWS end
>> Referring Doc:

https://www.reddit.com/r/aws/comments/an4s0o/having_issues_listing_all_files_in_an_s3_bucket/

Any ideas on how to properly configure the commvault backup?

**Comment by harshavardhana** [comment]
@kirihst can you test this with latest release?

**Comment by kirihst** [comment]
I think our issue persists for quiete a few releases now, was there a bug?
I'd have to wait for the next maintenance window, which could potentially take 2 weeks or a little longer. Anything else we could try?

**Comment by klauspost** [comment]
(converting to discussion, since it seems more like an application problem)

>  Anything else we could try?

There is no real information in what you posted. Maybe check the MinIO logs and see if there is anything in there.

The calls you are listing are internal to your application, so we can't see what it is actually doing.

If you could do an `mc admin trace -v play >reqs.txt` while it is running and `gzip reqs.txt` and upload it here we can see what exactly is failing.

**Comment by kirihst** [comment]
Hey klauspost, thank you for your answer. I'll get the trace tomorrow and give an update

**Comment by kirihst** [comment]
I got the trace appended, hope there's anything helpfull in there. If i can provide additional help, please let me know.
[reqs.txt.gz](https://github.com/user-attachments/files/16244888/reqs.txt.gz)

**Reply by klauspost** [reply, marked as answer]
> The authorization header is malformed; the region is wrong; expecting 'BOREUS'

Seems like the the application uses region `us-east-1`.

**Reply by kirihst** [reply]
how did you get to that conclusion? just want to comprehend. in our /etc/default file the region is defined as "MINIO_SITE_REGION=BOREUS", the error
>The authorization header is malformed; the region is wrong; expecting 'BOREUS'

to me looks like the application already expects the value to be just that. How would i find out that actually the region should be "us-east-1"? In commvault i can only set a server url, where would i go about changing the (expected) region?

thanks a bunch for your quick help

**Reply by klauspost** [reply]
The server expects `BOREUS`. The application sends `us-east-1` signature. Application should/could use GetBucketLocation, but seems like it doesn't.

Maybe don't change the region at all, though it seems very strange that the application can't change it.
